cmd/awscollector/Dockerfile

```diff
@@ -1,19 +1,59 @@
+# The binary source type (options: build, copy)
+ARG BIN_SRC_TYPE=build
+
 ################################
-#	Building Stage         #	
-#			       #	
+#	Certificate Stage      #
+#			       #
 ################################
-FROM alpine:latest as build 
+FROM alpine:latest AS certs
 
-# update cert
 RUN apk --update add ca-certificates
 
+################################
+#	Building Stage         #
+#			       #
+################################
+FROM golang:1.17 AS prep-build
+
+ENV GOPROXY=direct
+
+ARG TARGETARCH
+
+# download go modules ahead to speed up the building
+WORKDIR /workspace
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+# copy source
+COPY . .
+
+# build
+RUN make ${TARGETARCH}-build-only
+
+# move
+RUN mv /workspace/build/linux/$TARGETARCH/aoc /workspace/awscollector
+
+################################
+#	Copying Stage          #
+#			       #	
+################################
+FROM scratch AS prep-copy
+
 WORKDIR /workspace
 
 ARG TARGETARCH
 
 # copy artifacts
 # always assume binary is created
 COPY build/linux/$TARGETARCH/aoc /workspace/awscollector
+
+################################
+#	Packing Stage          #
+#			       #
+################################
+FROM prep-${BIN_SRC_TYPE} AS package
```

Comment:
Is this stage necessary?  Does this not work?

```DOCKERFILE
COPY --from=${BIN_SRC_TYPE} /workspace/awscollector /awscollector
COPY config.yaml /etc/otel-config.yaml
COPY config/ /etc/
```

Reply:
It seems necessary and that does not work.
```
Dockerfile:72
--------------------
  70 |     
  71 |     COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
  72 | >>> COPY --from="prep-${BIN_SRC_TYPE}" /workspace/awscollector /awscollector
  73 |     COPY config.yaml /workspace/config/otel-config.yaml
  74 |     COPY config/ /workspace/config/
--------------------
error: failed to solve: failed to parse stage name "prep-${BIN_SRC_TYPE}": invalid reference format: repository name must be lowercase
```

Reply:
booooooooooooooooo.